### PR TITLE
enhance(population): add population OMM

### PR DESCRIPTION
**Pull request by lucasrodes**
### Re-structure how we store our population OMM.
  - So far, it was only saved in `owid/latest/key_indicators` dataset. Now, it will live in `demography/<version>/population` along with all the recipe code to produce it.
  - `owid/latest/key_indicators` will still provide this OMM, as many datasets depend on it. It loads it from `demography/<version>/population`. This is transparent to the user.

### Add historical regions/countries to population OMM
  - Population for the following countries has been added to our population OMM: Czechoslovakia, East Germany, Eritrea and Ethiopia, Serbia and Montenegro, USSR, West Germany, Yemen Arab Republic, Yemen People's Republic, Yugoslavia
  - Note that these are only available via `demography/<version>/population` and not via `key_indicators`. Additional checks need to be done to ensure that these can safely be added to `key_indicators`.
  - Example usage:

    ```python
    from owid.catalog import Dataset
    from etl.paths import DATA_DIR
    
    ds = Dataset(DATA_DIR / "garden" / "demography" / "2022-12-08" / "population")
    tb = ds["population"]
    ```

## New data added:
- Data for the following countries prior to 1950: 'Isle of Man', 'Saint Martin (French part)', 'Guam', 'Tokelau', 'Cayman Islands', 'Gibraltar', 'Africa', 'Saint Helena', 'Saint Barthelemy', 'Northern Mariana Islands', 'Guernsey', 'British Virgin Islands', 'Kosovo', 'Macao', 'South Sudan', 'Wallis and Futuna', 'Bermuda', 'French Polynesia', 'Cook Islands', 'Jersey'
- Former countries added: 'Czechoslovakia','East Germany','Eritrea and Ethiopia','Serbia and Montenegro','USSR','West Germany','Yemen Arab Republic','Yemen People's Republic','Yugoslavia','Akrotiri and Dhekelia'

## Sanity checks
I have done sanity checks comparing values between old and new dataset (including affected `key_indicators` dataset) and the differences are within the expected range.

The major difference in population estimates comes from the fact that we are now reporting values for some countries prior to 1950 (listed above), and hence region aggregates (i.e. continents, income groups, World) add up to larger values. For instance, the addition of South Sudan leads to noticeable increases in Africa region. Same happens in Oceania, as we now report values for Cook Islands, Tokelau and many others prior to 1950.

### CODE REVIEW

**Review by pabloarosado — COMMENTED** [review]
I've made some comments, some of them may imply changes. I think we should visually do a comparison of all countries and regions populations, before merging. Let me know if you need help.

File: etl/steps/data/garden/demography/2022-12-08/population/Untitled.ipynb [hunk 1]
```diff
@@ -0,0 +1,434 @@
+{
```
[the remark this answers is not in the document]
Comment (lucasrodes) [reply etl/steps/data/garden/demography/2022-12-08/population/Untitled.ipynb:1]: mistake, deleting

File: etl/steps/data/garden/demography/2022-12-08/population/__init__.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    """Select adequate source for each country-year.
+
+    Rows are selected based on the following relevance scale: "unwpp" > "gapminder" > "hyde"
+    """
+    log.info("population: selecting source...")
+    df = df.loc[df.population > 0]
+
+    # If a country has UN data, then remove all non-UN data after 1949
+    has_un_data = set(df.loc[df.source == "unwpp", "country"])
+    df = df.loc[-((df.country.isin(has_un_data)) & (df.year >= 1950) & (df.source != "unwpp"))]
```
[the remark this answers is not in the document]
Comment (lucasrodes) [reply etl/steps/data/garden/demography/2022-12-08/population/__init__.py:106]: `~` and `-` are equivalent when applied to booleans AFAIK. I'll change it to `~` if we decide this is the common style we use.

File: etl/steps/data/garden/demography/2022-12-08/population/__init__.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        "Low-income countries",
+        "Lower-middle-income countries",
+        "Upper-middle-income countries",
+        "European Union (27)",
+    ]
+    # make sure to exclude regions if already present
+    df = df.loc[-df.country.isin(regions)]
+    # re-estimate regions
+    for region in regions:
+        df = geo.add_region_aggregates(df=df, region=region)
```
[the remark this answers is not in the document]
Comment (lucasrodes) [reply etl/steps/data/garden/demography/2022-12-08/population/__init__.py:149]: I am not following you on this one.

I am re-estimating the region aggregates as per our definitions here. This is independent of what the source does.

[the remark this answers is not in the document]
Comment (lucasrodes) [reply etl/steps/data/garden/demography/2022-12-08/population/__init__.py:149]: got it via discussing on slack

File: etl/steps/data/garden/demography/2022-12-08/population/__init__.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    tb = make_table()
+
+    # create dataset
+    log.info("population: create dataset")
+    ds = Dataset.create_empty(dest_dir)
+
+    # manage metadata
+    log.info("population: add metadata")
+    ds.metadata.update_from_yaml(METADATA_PATH)
+    tb.update_metadata_from_yaml(METADATA_PATH, "population")
```
[the remark this answers is not in the document]
Comment (lucasrodes) [reply etl/steps/data/garden/demography/2022-12-08/population/__init__.py:56]: nice one

**Comment by lucasrodes** [comment]
I have done visual and numerical checks, data seems in order! Added some comments to the description of the PR with some "known issues"

**Pull request merged**